core/arch/x86/encode.c

```diff
@@ -1096,7 +1095,7 @@ opnd_needs_evex(opnd_t opnd)
 
 static bool
 opnd_type_ok(decode_info_t *di /*prefixes field is IN/OUT; x86_mode is IN*/, opnd_t opnd,
-             int optype, opnd_size_t opsize)
+             int optype, opnd_size_t opsize, ushort flags)
```

Comment:
Measuring perf would be best, but my intuition says to not add parameters to this critical-path function that are not needed for 99%+ of calls.  Could we put the info needed inside di, like is done right after this code for the full encode for flags like REQUIRES_VEX.

Reply:
Done. I need the test in encoding_possible_pass, so we will read this for every instr_info_t we're checking. We're only saving N/avg_number_of_operands by moving it up.

Reply:
I don't see any change here.

I also don't follow the comment: why would the flags need to be read "for every instr_info_t" template entry?  It would be read just once per instruction being encoded, while saving tons of parameter passings.

Reply:
Sorry change was still in my local sandbox. Will add.
The flag is a property of the instr_info_t entry. What I meant was that we will read it for every instr_info_t we're testing (one or more per encoded instruction). So by factoring it out from TEST_OPND/opnd_type_ok to encoding_possible_pass, we're saving passing it number of calls of TEST_OPND.

Reply:
Done.